File: 2022/zig/day1/src/day1p1.zig
const std = @import("std");

const lib = @import("lib.zig");

//const ArrayList = std.ArrayList;
const stdout = std.io.getStdOut().writer();
//var gpa = std.heap.GeneralPurposeAllocator(.{}){};
//const allocator = gpa.allocator();

pub fn main() !void {
    var list = try lib.get_elf_calories("input");

    if (list.items.len > 0) {
        try stdout.print("{d}", .{list.items[0]});
    }

    list.deinit();
}

File: 2022/cpp/day1/src/day1p1.cpp
#include "lib.h"

#include <iostream>

int main(void) {
    auto elf_cals = get_elf_calories("input");

    if (!elf_cals.empty()) {
        std::cout << elf_cals[0] << "\n";
    } else {
        std::cout << "Error: no calories" << "\n";
    }

    return 0;
}
